File: src/main/java/com/github/kalheeso/provax/domain/Vacina.java
package com.github.kalheeso.provax.domain;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Data;
import lombok.Setter;

@Entity
public class Vacina {
    @Setter(onMethod_ = @Deprecated)
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String titulo;
    private String descricao;
    //Quantidade de doses
    private int doses;
    // 1 - dia, 2 - semana, 3 - mes, 4 - ano
    private int periodicidade;
    private int intervalo;
}

File: frontend/src/app/vacine/core/entities/Vacina.ts
export default class Vacina {
  id!: number
  titulo!: string
  doses!: number
  descricao!: string
  periodicidade!: string
  intervalo!: number

  constructor(id : number = 0, titulo: string = '', doses: number = 0, descricao: string = '', periodicidade: string = '', intervalo: number = 0) {
    this.id = id;
    this.titulo = titulo;
    this.doses = doses;
    this.descricao = descricao;
    this.periodicidade = periodicidade;
    this.intervalo = intervalo;
  }

}
